wagtail/users/tests/test_admin_views.py

```diff
@@ -474,7 +506,6 @@ def test_create(self):
     )
     @override_settings(
         WAGTAIL_USER_CREATION_FORM="wagtail.users.tests.CustomUserCreationForm",
-        WAGTAIL_USER_CUSTOM_FIELDS=["country", "document"],
     )
     def test_create_with_custom_form(self):
```

Comment:
These view tests that override `WAGTAIL_USER_CREATION_FORM`/`WAGTAIL_USER_EDIT_FORM` are unfortunately bogus...

I removed the `WAGTAIL_USER_CUSTOM_FIELDS` override for the same reason as my comment https://github.com/wagtail/wagtail/pull/11952#discussion_r1598646882 (the `CustomUserCreationForm` already defines both fields), and the test still passes. However, it's not because we are setting `WAGTAIL_USER_CREATION_FORM`, but rather because we have `WAGTAIL_USER_CUSTOM_FIELDS` defined in the settings directly:

https://github.com/wagtail/wagtail/blob/7fca2d74a04f8aa6a796a7f645215d06e272208b/wagtail/test/settings.py#L208-L212

Which, looking at the comment, is fair enough.

However, I'm pretty sure that unfortunately the `WAGTAIL_USER_CREATION_FORM` and `WAGTAIL_USER_EDIT_FORM` settings can no longer be tested with `override_settings` for the same reason after the view was refactored to CBV, as the `get_user_{create,edit}_form()` function is called on the class-level e.g. in https://github.com/wagtail/wagtail/commit/61a5e197cd990efdd7993470ff01a8f0b4ae3c3c#diff-f869d9c6b0a92ab5be8bcd0bff8dbc3695520bd183b9326bd66fc12ecdf840c0R184

We could've overridden `get_form_class()` and call the `get_user_{create,edit}_form()` function there instead of setting `form_class` directly, but this would've been reverted anyway when we adopted viewsets in #11823. This is because the viewset's `get_form_class()` is called at import time (to be passed in `as_view()`).

Unfortunately I don't know a good way to test such viewset customisations... So far we've got away with defining multiple viewsets for different customisation options, but for built-in viewsets like this it's quite difficult. Even if we use mocks, we can only get as far as the viewset and not the view itself, at least not without making it so ugly. Which means there's no practical way to replace this test once we fully remove these settings in Wagtail 7.0, or at least not that I know of.

Which, I assume was the reason why the test in https://github.com/wagtail/wagtail/commit/9dda314263eb69fb72fbe410435ee66f01a90c08#diff-ac75c762cbb5c68c3a0f0294869f5082ae25f7edd7aef4e2f12895b1f674e36fR1588-R1615 only covered setting the `AppConfig`'s attribute, and not actually testing that the view indeed uses a custom form class.